### ISSUE TITLE
Check if all case statements are on some indent in PSR2.ControlStructures.SwitchDeclaration

### ISSUE THREAD

**Issue by umherirrender** [squizlabs/PHP_CodeSniffer#3353]
**Describe the bug**
Each `case` of the same `switch` should have the same indent as the previous one to have the cases proper aligned.

**Code sample**
```php
$test = 2;

switch ( $test ) {
	case 2:
		echo "test 2";
		break;
	case 3:
		echo "test 3";
		break;
	 case 4:
		 echo "test 4";
		 break;
}
```

**Custom ruleset**
```xml
<?xml version="1.0"?>
<ruleset name="My Custom Standard">
  <rule ref="PSR2.ControlStructures.SwitchDeclaration" />
</ruleset>
```

**To reproduce**
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. See error message displayed
```
No report
```

**Expected behavior**
A report that the `case 4` is not on the same indent as the other cases

**Versions (please complete the following information):**
 - OS: Windows 10
 - PHP: 8.0
 - PHPCS: 3.6.0
 - Standard: -

**Comment by jrfnl** [comment]
IMO that's not the concern of the `PSR2.ControlStructures.SwitchDeclaration` sniff. The PSR2 standard includes the `Generic.WhiteSpace.ScopeIndent` sniff which reports on the incorrect indent when given the above code sample:

```
 12 | ERROR | [x] Line indented incorrectly; expected 4 spaces, found 5 (Generic.WhiteSpace.ScopeIndent.IncorrectExact)
```

**Comment by umherirrender** [comment]
My problem was, that 
```php
$test = 2;

switch ( $test ) {
	case 2:
		echo "test 2";
		break;
	case 3:
		echo "test 3";
		break;
	 case 4:
		echo "test 4";
		break;
}
```
was fixed to
```php
$test = 2;

switch ( $test ) {
	case 2:
		echo "test 2";
		break;
	case 3:
		echo "test 3";
		break;
	 case 4:
		echo "test 4";
		 break;
}
```
It takes a bit for me to spot the problem, because the case body before the problematic case was very long and it was not easy to see the extra space.

The scope indent sniff would help here, but than also the question if `case` should be indent once inside the `switch` or on the same level as the `switch`. With checking the alignment of all `case` statements, the question about how deep the indent should be is not needed.

But I am okay if this gets declined/invalid/wontfix.

**Comment by gsherwood** [comment]
If you include only `PSR2.ControlStructures.SwitchDeclaration` then your code isn't going to get fixed as outlined in your previous comment, so something else was obviously included in the standard to cause that.

But as @jrfnl said, the `Generic.WhiteSpace.ScopeIndent` does a lot of the heavy lifting when it comes to indenting all sorts of control structures, including case statements. The `PSR2.ControlStructures.SwitchDeclaration` sniff, and most others, don't include code for checking indents for control structures because it's so complex to figure out. They leave it to the ScopeIdent sniff itself.

Not that the ScopeIndent sniff can be told that some scopes are non-indenting. For example, the included PEAR coding standard enforces that CASE statements are indented to the same level as the SWITCH keyword by including a custom sniff that sets the `$nonIndentingScopes` member var to include `T_SWITCH`, but this is not available as a sniff property so it can't be set inside a ruleset.xml file.

Does all that make sense?
